### DISCUSSION TITLE
vim support

### DISCUSSION THREAD

**Discussion by Song-Tianxiang** [kakoune-lsp/kakoune-lsp#699] [General]
Hello maintainers, thank you for developing such a great lsp client.

As a long-time `vim` user and new to rust, I would like to know if it is possible to have `kak-lsp` support vim since kak-lsp runs as a standalone binary. On the one hand, it seems that `vim` does not lack the means of communicating with `kak-lsp` that `kakoune` has. On the other hand, the existing vim lsp client does not perform very well. Maybe rust can improve this?

I'm wonder it just requires some extra vim scripts, or is there a lot of extra work we need to do all over again. I'd like to understand the possibilities and the likely workload.

**Comment by krobelus** [comment]
consider switching to Kakoune, I'm pretty sure it can do everything you want (minus the Git integration but that's getting better, also you can use Tig)

>  On the other hand, the existing vim lsp client does not perform very well.

would be interesting to know why. Probably neovim works fine, they have built-in one.

> On the one hand, it seems that vim does not lack the means of communicating with kak-lsp

what's the Vim equivalent to `echo do-something | kak -p`? Also it would need some way to extract buffer state like selection coordinates.
If someone wants to add vim support, they could start experimenting on a fork of kak-lsp. But I think there's already plenty of vim LSP clients, I'd be surprised if none of them is usable.
